index.js

```diff
@@ -103,36 +164,58 @@ try {
 
   // For the POST request, the above content is set as file contents for the
   // given filename.
-  const request = JSON.stringify({
-    files: {[core.getInput('filename')]: {content: JSON.stringify(content)}}
-  });
-
-  // Perform the actual request. The user agent is required as defined in
-  // https://developer.github.com/v3/#user-agent-required
-  const req = http.request(
-      {
-        host: 'api.github.com',
-        path: '/gists/' + core.getInput('gistID'),
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'Content-Length': request.length,
-          'User-Agent': 'Schneegans',
-          'Authorization': 'token ' + core.getInput('auth'),
+  const request =
+      JSON.stringify({files: {[filename]: {content: JSON.stringify(content)}}});
+
+  // If "forceUpdate" is set to true, we can simply update the gist. If not, we have to
+  // get the gist data and compare it to the new value before.
+  if (core.getBooleanInput('forceUpdate')) {
+    updateGist(request);
+
+  } else {
+
+    // Get the old gist.
+    const getGistOptions = {
+      host: 'api.github.com',
+      path: '/gists/' + core.getInput('gistID'),
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'User-Agent': 'Schneegans',
+        'Authorization': 'token ' + core.getInput('auth'),
+      }
+    };
+
+    doRequest(getGistOptions, JSON.stringify({})).then(oldGist => {
+      if (oldGist.statusCode < 200 || oldGist.statusCode >= 400) {
+        // print the error, but don't fail the action.
+        console.log(
```

Comment:
@Schneegans because every run of action will go over this request, I think it should be changed from `console.log` to `core.setFailed`.  In case the user will enter the wrong `gistId`, it should fail.
WDYT?

Reply:
Yeah, you're right, it could fail early at this point. However, it will fail anyways when the action will later try to update the gist.